File: tools/python-execute-eval/agent-solutions/claude-sonnet-4-6/search-then-read.mjs
import { kody } from 'kody:runtime'

export default async function main(params) {
	const results = await kody.call('search', { query: params.query })
	const hits = results.hits ?? []
	if (hits.length === 0) throw new Error('No results found')
	return kody.call('notes.get', { id: hits[0].id })
}

File: tools/python-execute-eval/agent-solutions/claude-sonnet-4-6/search-then-read.py
async def main(params):
    results = await kody.call("search", {"query": params["query"]})
    hits = results.get("hits", [])
    if not hits:
        raise RuntimeError("No results found")
    note = await kody.call("notes.get", {"id": hits[0]["id"]})
    return note
